File: v-im-pc-2025/src/renderer/src/message/Message.ts
export interface Message<T = any> {
  //消息id,雪花id,有序增长
  id?: string
  //消息文件类型
  messageType: string
  //聊天室id
  chatId: string
  //消息发送人
  fromId: string
  //是否是本人
  mine?: boolean
  //消息内容
  content: string
  //消息时间
  timestamp?: number
  //消息类型：私聊|群聊
  chatType: 'friend' | 'group'
  //扩展
  extend?: T
  //是否显示通知
  notification?: boolean
}

File: v-im-server-2025/v-im-server/src/main/java/com/vim/tio/messages/Message.java
package com.vim.tio.messages;

import cn.hutool.json.JSONObject;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Field;

import java.io.Serial;
import java.io.Serializable;

/**
 * websocket 通讯的消息类型
 *
 * @author 乐天
 * @since 2018-10-07
 */
@Data
public class Message implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;
    /**
     * 消息id
     */
    @Id
    private String id;

    /**
     * 消息的来源ID（如果是私聊，则是用户id，如果是群聊，则是群组id）
     */
    @Field("chat_id")
    private String chatId;

    /**
     * 聊天室类型 friend|group
     */
    @Field
    private String chatType;

    /**
     * 消息类型 文本|附件|其他
     */
    @Field("message_type")
    private String messageType;

    /**
     * 消息内容
     */
    @Field("content")
    private String content;

    /**
     * 消息的发送者id
     */
    @Field("from_id")
    private String fromId;

    /**
     * 服务端时间戳毫秒数
     */
    @Field("timestamp")
    private Long timestamp;

    /**
     * 扩展字段，格式化为json
     */
    @Field("extend")
    private JSONObject extend;


    /**
     * 消息的发送者id
     */
    @Field("chat_key")
    private String chatKey;

}
